compilation warnings

Hi. 

I get the following warnings
```
In file included from /nfs/OPTI/bzfserra/ZIB/scip2/tests/Criterion-install/include/criterion/criterion.h:119:0,
                 from /nfs/OPTI/bzfserra/ZIB/scip2/tests/include/scip_test.h:15,
                 from /nfs/OPTI/bzfserra/ZIB/scip2/tests/src/cons/enforelaximplemented.c:24:
/nfs/OPTI/bzfserra/ZIB/scip2/tests/Criterion-install/include/criterion/internal/test.h:71:13: error: redundant redeclaration of ‘cri_asserts_passed_incr’ [-Werror=redundant-decls]
 CR_API void cri_asserts_passed_incr(void);
             ^
In file included from /nfs/OPTI/bzfserra/ZIB/scip2/tests/Criterion-install/include/criterion/assert.h:1679:0,
                 from /nfs/OPTI/bzfserra/ZIB/scip2/tests/Criterion-install/include/criterion/criterion.h:32,
                 from /nfs/OPTI/bzfserra/ZIB/scip2/tests/include/scip_test.h:15,
                 from /nfs/OPTI/bzfserra/ZIB/scip2/tests/src/cons/enforelaximplemented.c:24:
/nfs/OPTI/bzfserra/ZIB/scip2/tests/Criterion-install/include/criterion/internal/assert.h:79:13: note: previous declaration of ‘cri_asserts_passed_incr’ was here
 CR_API void cri_asserts_passed_incr(void);
```

and

```
In file included from /nfs/OPTI/bzfserra/ZIB/scip2/tests/Criterion-install/include/criterion/internal/assert.h:44:0,
                 from /nfs/OPTI/bzfserra/ZIB/scip2/tests/Criterion-install/include/criterion/assert.h:1679,
                 from /nfs/OPTI/bzfserra/ZIB/scip2/tests/Criterion-install/include/criterion/criterion.h:32,
                 from /nfs/OPTI/bzfserra/ZIB/scip2/tests/include/scip_test.h:15,
                 from /nfs/OPTI/bzfserra/ZIB/scip2/tests/src/cons/enforelaximplemented.c:24:
/nfs/OPTI/bzfserra/ZIB/scip2/tests/Criterion-install/include/criterion/internal/../hooks.h:50:1: error: function declaration isn’t a prototype [-Werror=strict-prototypes]
 typedef void (*f_report_hook)();
```

Both happen in bleeding and master. I am running on a linux machine (ubuntu 16.04)

..../include/criterion/internal/../hooks.h:50:1: error: function declaration isn’t a prototype [-Werror=strict-prototypes]
 typedef void (*f_report_hook)();
Can this be fixed?
Just add void...
typedef void (*f_report_hook)(void);

--- a/include/criterion/internal/test.h
+++ b/include/criterion/internal/test.h
@@ -26,6 +26,7 @@
 
 #include "designated-initializer-compat.h"
 #include "common.h"
+#include "assert.h"
 
 #ifdef __OBJC__
 # import <Foundation/Foundation.h>
@@ -68,7 +69,6 @@ CR_BEGIN_C_API
 CR_API void criterion_internal_test_setup(void);
 CR_API void criterion_internal_test_main(void (*fn)(void));
 CR_API void criterion_internal_test_teardown(void);
-CR_API void cri_asserts_passed_incr(void);
 
 CR_END_C_API
 

